### PR TITLE
Added reserve method to RefToBaseVector

### LINKED ISSUE #123: RefToBaseVector does not implement reserve
See https://github.com/cms-sw/cmssw/issues/33459

### LINKED ISSUE #33459: RefToBaseVector does not implement reserve
it also implement a ```vectorHolder()``` method of dubious semantics.

 trackRefs.vectorHolder()->reserve(mysize); //  NOT a good idea

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:
Added reserve method to RefToBaseVector

#### PR validation:

Code compiles.

fixes #33459
fixes makortel/framework#123

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: DataFormats/Common/interface/RefToBaseVector.h [hunk 1]
```diff
@@ -161,6 +161,12 @@ namespace edm {
     return holder_ ? holder_->size() : 0;
   }
 
+  template <class T>
+  inline void RefToBaseVector<T>::reserve(size_type n) {
+    if (holder_ != nullptr)
+      holder_->reserve(n);
```
Comment (makortel) [inline DataFormats/Common/interface/RefToBaseVector.h:167]: This means that `reserve()` in 
```cpp
RefToBaseVector<T> foo;
foo.reserve(N);
```
is a no-op, right?

(I'm guessing @VinInn's use case to be
```cpp
        edm::RefToBaseVector<reco::Track> trackRefs;
        // trackRefs.vectorHolder()->reserve(trackCollection.size());  NOT a good idea
```
in https://github.com/cms-sw/cmssw/pull/33462/files#diff-509b7e4ef9898b7d529699031d792fa528489903211d78de7eb1e088736427ccR804-R805
)

Comment (VinInn) [reply DataFormats/Common/interface/RefToBaseVector.h:167]: my understanding is that the semantic of the class is to create the holder if not found...
So the proposed mod may be no-op....

**Comment by Dr15Jones** [comment]
A solution is more complicated and would require much more thought.

**Pull request closed without merging**